cpp/tests/interop/to_arrow_host_test.cpp

```diff
@@ -0,0 +1,953 @@
+  auto list_arr =
+    get_nanoarrow_list_array<int64_t>({1, 2, 3, 4, 5, 6, 7, 8, 9}, {0, 2, 4, 5, 6, 7, 9});
+  std::vector<int32_t> offset{0, 3, 4, 6};
+
+  nanoarrow::UniqueArray expected_arr;
+  NANOARROW_THROW_NOT_OK(
+    ArrowArrayInitFromSchema(expected_arr.get(), expected_schema.get(), nullptr));
+  expected_arr->length = input_view.num_rows();
+
+  auto array_a        = expected_arr->children[0];
+  auto view_a         = input_view.column(0);
+  array_a->length     = view_a.size();
+  array_a->null_count = view_a.null_count();
+
+  str_array.move(array_a->children[0]);
+  int_array.move(array_a->children[1]);
+  bool_array.move(array_a->children[2]);
+
+  array_a->children[3]->length     = input_view.num_rows();
+  array_a->children[3]->null_count = 0;
+
+  auto offset_buf = ArrowArrayBuffer(array_a->children[3], 1);
+  EXPECT_EQ(
+    NANOARROW_OK,
+    ArrowBufferAppend(
+      offset_buf, reinterpret_cast<void const*>(offset.data()), offset.size() * sizeof(int32_t)));
+  list_arr.move(array_a->children[3]->children[0]);
+
+  ArrowBitmap mask;
+  ArrowBitmapInit(&mask);
+  NANOARROW_THROW_NOT_OK(ArrowBitmapReserve(&mask, 3));
+  NANOARROW_THROW_NOT_OK(ArrowBitmapAppend(&mask, 1, 2));
+  NANOARROW_THROW_NOT_OK(ArrowBitmapAppend(&mask, 0, 1));
+
+  auto array_struct = array_a->children[4];
+  auto view_struct  = view_a.child(4);
+  ArrowArraySetValidityBitmap(array_struct, &mask);
+  array_struct->null_count = view_struct.null_count();
+  array_struct->length     = view_struct.size();
+
+  str2_array.move(array_struct->children[0]);
+  int2_array.move(array_struct->children[1]);
+
+  NANOARROW_THROW_NOT_OK(ArrowArrayFinishBuildingDefault(expected_arr.get(), nullptr));
+
+  auto got_arrow_host = cudf::to_arrow_host(input_view);
+  EXPECT_EQ(ARROW_DEVICE_CPU, got_arrow_host->device_type);
+  EXPECT_EQ(-1, got_arrow_host->device_id);
+  EXPECT_EQ(nullptr, got_arrow_host->sync_event);
+
+  ArrowArrayView expected, actual;
+  NANOARROW_THROW_NOT_OK(ArrowArrayViewInitFromSchema(&expected, expected_schema.get(), nullptr));
+  NANOARROW_THROW_NOT_OK(ArrowArrayViewSetArray(&expected, expected_arr.get(), nullptr));
+
+  NANOARROW_THROW_NOT_OK(ArrowArrayViewInitFromSchema(&actual, expected_schema.get(), nullptr));
+  NANOARROW_THROW_NOT_OK(ArrowArrayViewSetArray(&actual, &got_arrow_host->array, nullptr));
+  compare_arrays(&expected, &actual);
+  ArrowArrayViewReset(&actual);
+
+  got_arrow_host = cudf::to_arrow_host(input_view.column(0));
+  NANOARROW_THROW_NOT_OK(
+    ArrowArrayViewInitFromSchema(&actual, expected_schema->children[0], nullptr));
+  NANOARROW_THROW_NOT_OK(ArrowArrayViewSetArray(&actual, &got_arrow_host->array, nullptr));
+  compare_arrays(expected.children[0], &actual);
+  ArrowArrayViewReset(&actual);
+
+  ArrowArrayViewReset(&expected);
+}
+
+template <typename T>
+using fp_wrapper = cudf::test::fixed_point_column_wrapper<T>;
+
+TEST_F(ToArrowHostDeviceTest, FixedPoint64Table)
```

Comment:
Can we also add a test with `FixedPoint32Table` as well if not too much overhead?